lib/robots/dor_repo/etd_submit/other_metadata.rb

```diff
@@ -11,51 +11,55 @@ def initialize(opts = {})
 
-          druid_tools_druid = DruidTools::Druid.new(druid, Settings.sdr.local_workspace_root)
-          content_dir = druid_tools_druid.content_dir
+          object = DruidTools::Druid.new(druid, Settings.sdr.local_workspace_root)
+          content_dir = object.content_dir
 
           # now transfer the pdfs from lyberapps into the workspace parent content directory
           source_dir = File.join(ETD_WORKSPACE, druid)
           transfer_object(source_dir, content_dir)
 
           return if etd.nil?
 
-          populate_datastream(etd, 'contentMetadata')
-          populate_datastream(etd, 'rightsMetadata')
-          populate_datastream(etd, 'identityMetadata')
-          populate_datastream(etd, 'versionMetadata')
+          create_metadata(etd, 'contentMetadata', druid)
+          create_metadata(etd, 'rightsMetadata', druid)
+          create_metadata(etd, 'identityMetadata', druid)
+          create_metadata(etd, 'versionMetadata', druid)
         end
 
-        # create a datastream in the repository for the given etd object
-        def populate_datastream(etd, ds_name)
-          metadata = case ds_name
+        private
+
+        # create metadata in the repository for the given etd object
+        def create_metadata(etd, md_name, druid)
+          metadata = case md_name
                      when 'identityMetadata' then Dor::Etd::IdentityMetadataGenerator.generate(etd)
                      when 'contentMetadata' then Dor::Etd::ContentMetadataGenerator.generate(etd)
                      when 'rightsMetadata' then Dor::Etd::RightsMetadataGenerator.generate(etd)
                      when 'versionMetadata' then Dor::Etd::VersionMetadataGenerator.generate(etd.pid)
                      end
           return if metadata.nil?
 
-          populate_datastream_in_repository(etd, ds_name, metadata)
+          create_legacy_metadata(druid, metadata_symbol(md_name), metadata)
         end
 
-        # create a datastream for the given etd object with the given datastream name, label, and metadata blob
-        def populate_datastream_in_repository(etd, ds_name, metadata)
-          label = case ds_name
-                  when 'identityMetadata' then 'Identity Metadata'
-                  when 'contentMetadata' then 'Content Metadata'
-                  when 'rightsMetadata' then 'Rights Metadata'
-                  when 'versionMetadata' then 'Version Metadata'
-                  else ''
-                  end
-          attrs = { mimeType: 'application/xml', dsLabel: label, content: metadata }
-          datastream = ActiveFedora::Datastream.new(etd.inner_object, ds_name, attrs)
-          datastream.controlGroup = 'M'
-          datastream.save
+        def create_legacy_metadata(druid, md_sym, new_metadata)
+          object_client = Dor::Services::Client.object(druid)
+
+          # legacy_update will create the metadata
+          object_client.metadata.legacy_update(
+            md_sym => {
```

Comment:
This looks good. one optimization here is that we can generate all of the different content and pass them in a single `legacy_update` request.  e.g.:

```ruby
object_client.metadata.legacy_update(
  content: {
    updated: Time.now,
    content: new_metadata
  },
  identity: {
    ...
  },
  version: {
    ...
  },
  rights: {
    ...
  })

See http://sul-dlss.github.io/dor-services-app/#operation/metadata#update_legacy_metadata
```

Reply:
will update